### PR TITLE
dev vp-372 Move export module Product data json export to v2

### PR DESCRIPTION
**Pull request by AliveMen**
add generic export infrastructure for product export

### CODE REVIEW

File: VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportPagedDataSource.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            var result = Array.Empty<CatalogProduct>();
+            int totalCount = 0;
+
+            var responseGroup = BuildResponseGroup();
+            var productIds = searchCriteria.ObjectIds?.ToArray() ?? Array.Empty<string>();
+
+            if (productIds.IsNullOrEmpty())
+            {
+                var catalogSearchCriteria = AbstractTypeFactory<SearchCriteria>.TryCreateInstance();
+                catalogSearchCriteria.SearchInChildren = true;
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportPagedDataSource.cs]: Shouldn't it go from `searchCriteria`?

File: VirtoCommerce.CatalogModule.Data/ExportImport/ExportableProduct.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
+using System.Linq;
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/ExportImport/ExportableProduct.cs]: Please move it to Catalog.Core (as here https://github.com/VirtoCommerce/vc-platform-core/commit/0429ecf54bd651826061cff8cb0adcf613e6b38f)

File: VirtoCommerce.CatalogModule.Data/ExportImport/TabularProduct.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
+using System;
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/ExportImport/TabularProduct.cs]: Please move it to Catalog.Core (as here https://github.com/VirtoCommerce/vc-platform-core/commit/0429ecf54bd651826061cff8cb0adcf613e6b38f)

File: VirtoCommerce.CatalogModule.Data/packages.config [hunk 4]
```diff
@@ -10,8 +10,11 @@
...
   <package id="System.ComponentModel.Primitives" version="4.3.0" targetFramework="net461" />
+  <package id="System.ValueTuple" version="4.4.0" targetFramework="net461" />
   <package id="valueinjecter" version="2.3.3" targetFramework="net451" />
-  <package id="VirtoCommerce.Domain" version="2.25.24" targetFramework="net461" />
+  <package id="VirtoCommerce.Domain" version="2.25.26" targetFramework="net461" />
+  <package id="VirtoCommerce.ExportModule.Core" version="2.1.1" targetFramework="net461" />
+  <package id="VirtoCommerce.ExportModule.Data" version="2.1.1" targetFramework="net461" />
   <package id="VirtoCommerce.Platform.Core" version="2.13.42" targetFramework="net461" />
-  <package id="VirtoCommerce.Platform.Data" version="2.13.37" targetFramework="net461" />
+  <package id="VirtoCommerce.Platform.Data" version="2.13.39" targetFramework="net461" />
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/packages.config]: Seems
id="VirtoCommerce.Platform.Core" version="2.13.42"
id="VirtoCommerce.Platform.Data" version="2.13.39"
should be synchronized, probably need to upgrade both 42 (and in project file)

File: VirtoCommerce.CatalogModule.Web/module.manifest [hunk 5]
```diff
@@ -2,9 +2,10 @@
 <module>
     <id>VirtoCommerce.Catalog</id>
     <version>2.23.21</version>
-    <platformVersion>2.13.42</platformVersion>
+    <platformVersion>2.13.39</platformVersion>
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/module.manifest]: I think we cannot downgrade it, maybe there are some deps on 42 version.

File: VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportPagedDataSource.cs (hunk 1 above)
Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportPagedDataSource.cs]: Criteria and Catalog filters work onliy when SearchInChildren is true. So when we open export blade from catalog module and SearchInChildren is false as default results are always empty. For better UI experience we should set SearchInChildren to true.

File: VirtoCommerce.CatalogModule.Core/VirtoCommerce.CatalogModule.Core.csproj [hunk 6]
```diff
@@ -0,0 +1,68 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Core/VirtoCommerce.CatalogModule.Core.csproj]: I forgot we do not have Core project for catalog in v2. Seems we do not need to add it then.
So let's move them back.

File: VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportDataQuery.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+using VirtoCommerce.ExportModule.Core.Model;
+
+namespace VirtoCommerce.CatalogModule.Data.ExportImport
+{
+    public class ProductExportDataQuery : ExportDataQuery
+    {
+        public bool SearchInVariations { get; set; }
+        public bool SearchInChildren { get; set; }
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/ExportImport/ProductExportDataQuery.cs]: Do we need it here, as it is not used now in criteria?

**Pull request merged**